xbmc/cores/RetroPlayer/RetroPlayerAutoSave.h

```diff
@@ -0,0 +1,46 @@
+/*
+ *      Copyright (C) 2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this Program; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+#include "threads/Thread.h"
+
+class IPlayer;
+
+namespace KODI
+{
+namespace RETRO
+{
+  class CRetroPlayerAutoSave : protected CThread
+  {
+  public:
+    CRetroPlayerAutoSave(IPlayer *player);
+
+    virtual ~CRetroPlayerAutoSave() override;
```

Comment:
You don't need the `virtual` here and IIR there are compilers who spit out a warning if you combine `virtual` and `override` for a method definition.